lib/gui/app/models/files.js

```diff
@@ -143,26 +144,13 @@ exports.splitPath = (fullpath, subpaths = []) => {
 /**
  * @summary Get constraint path device
  * @param {String} pathname - device path
- * @param {Function} callback - callback(error, constraintDevice)
+ * @returns {Drive} drive - drive object
  * @example
- * files.getConstraintDevice('/dev/disk2', (error, device) => {
- *   // ...
- * })
+ * const device = files.getConstraintDevice('/dev/disk2')
  */
-exports.getConstraintDevice = (pathname, callback) => {
-  drivelist.list((error, devices) => {
-    if (error) {
-      callback()
-      return
-    }
-
-    const constraintDevice = devices.find((device) => {
-      return device.device === pathname ||
-        device.devicePath === pathname
-    })
-
-    callback(null, constraintDevice)
-  })
+exports.getConstraintDevice = (pathname) => {
+  // This supposes the drive scanner is ready
```

Comment:
Why "supposes"? I'd rather it be sure about it, what happens if not?

Reply:
If it's not ready, it'll return `undefined`.
This is called when you open the experimental file picker so the scanner is already started and ready at this point.